perftest/spreadtestndall.cpp

```diff
@@ -121,155 +119,144 @@ int main(int argc, char *argv[])
+    const auto tol = (FLT)(10.0 * pow(10.0, -digits));
+    printf("digits=%d, tol = %.3g\n", digits, (double)tol);
 
-    // spread a single source, only for reference accuracy check...
-    opts.spread_direction = 1;
+    // set finufft options from CLI choices
+    finufft_opts fopts;
+    FINUFFT_DEFAULT_OPTS(&fopts);
+    fopts.upsampfac          = upsampfac;
+    fopts.spread_kerevalmeth = kerevalmeth;
+    fopts.spread_sort        = sort;
+    fopts.spread_kerpad      = kerpad;
+    fopts.debug              = debug;
+    fopts.showwarn           = 1;
+    fopts.spreadinterponly   = 1;
 
-    d_nonuniform[0] = 1.0;
-    d_nonuniform[1] = 0.0;            // unit strength
-    kx[0] = ky[0] = kz[0] = PI / 2.0; // at center
-    int ier = spreadinterp(N, N2, N3, d_uniform.data(), 1, kx.data(), ky.data(),
-                           kz.data(), d_nonuniform.data(),
-                           opts); // vector::data officially C++11 but works
+    FINUFFT_PLAN plan{};
+    BIGINT nmodes[3] = {N, N, N};
+    int ier          = FINUFFT_MAKEPLAN(dir, d, nmodes, +1, 1, tol, &plan, &fopts);
+    if (ier > 1) {
+      printf("error when creating the plan (ier=%d)!\n", ier);
+      return ier;
+    }
+
+    ier = FINUFFT_SETPTS(plan, M, kx.data(), ky.data(), kz.data(), 0, nullptr, nullptr,
+                         nullptr);
     if (ier != 0) {
-      printf("error when spreading M=1 pt for ref acc check (ier=%d)!\n", ier);
+      printf("error when setting NU pts (ier=%d)!\n", ier);
+      FINUFFT_DESTROY(plan);
       return ier;
     }
-    FLT kersumre = 0.0, kersumim = 0.0; // sum kernel on uniform grid
-    for (BIGINT i = 0; i < Ng; ++i) {
-      kersumre += d_uniform[2 * i];
-      kersumim += d_uniform[2 * i + 1]; // in case the kernel isn't real!
+
+    d_nonuniform.assign(M, CPX(0.0, 0.0));
+    d_uniform.assign(Ng, CPX(0.0, 0.0));
+    kx[0] = ky[0] = kz[0] = (FLT)PI / 2.0; // NU point location (unchanged)
+    auto kersum           = CPX(0.0, 0.0);
+    if (dir == 1) {
+      // type-1 (NU -> U): place unit at NU point, spread to grid and sum grid
+      d_nonuniform[0] = CPX(1.0, 0.0);
+      ier             = FINUFFT_EXECUTE(plan, d_nonuniform.data(), d_uniform.data());
+      if (ier != 0) {
+        printf("error when spreading M=1 pt for ref acc check (ier=%d)!\n", ier);
+        FINUFFT_DESTROY(plan);
+        return ier;
+      }
+      kersum = std::accumulate(d_uniform.begin(), d_uniform.end(), CPX(0.0, 0.0));
+    } else {
+      // type-2 (U -> NU): set uniform grid to ones, interpolate to NU point
+      for (BIGINT i = 0; i < Ng; ++i) d_uniform[i] = CPX(1.0, 0.0);
+      ier = FINUFFT_EXECUTE(plan, d_nonuniform.data(), d_uniform.data());
+      if (ier != 0) {
+        printf("error when interpolating for ref acc check (ier=%d)!\n", ier);
+        FINUFFT_DESTROY(plan);
+        return ier;
+      }
+      // result at the single NU point is the kernel sum
+      kersum = d_nonuniform[0];
```

Comment:
Seems the logic of testing error is changed for dir=2 compared to original spreadtestndall.cpp.
Now spreadtestnd.cpp and spreadtestndall.cpp have different logics too.